### PR TITLE
Buy album

### PR DESCRIPTION
**Pull request by raparicio6**
## Summary

A user can buy an album that has not been sold to another user. Also the user can not buy the same album more than once.
It is required to be authenticated.

## Trello Card

https://trello.com/c/K6t1TZlP

### CODE REVIEW

File: app/controllers/albums.js [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+      const filter = {
+        id: req.params.id
+      };
+
+      return albumService
+        .getLocalAlbumsByFilter(filter)
+        .then(albums => {
+          const [localAlbum] = albums;
+
+          if (localAlbum && localAlbum.userId === req.body.userId) throw errors.userAlreadyHasTheAlbum;
```
Comment (igna92ts) [inline app/controllers/albums.js:26]: cant the buy more than one??

File: app/controllers/albums.js [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+
+          if (localAlbum) throw errors.albumAlreadyHasOwnerUser;
```
Comment (igna92ts) [inline app/controllers/albums.js:28]: cant a same album be bought by more than one person??

File: app/controllers/albums.js [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
+
+          // The album is not in the database, so now it is saved
+          album.userId = req.body.userId;
+          return albumService.createAlbum(album);
```
Comment (igna92ts) [inline app/controllers/albums.js]: here you can just do
```
return albumService.createAlbum({ ...album, userId: req.body.userId });
```

File: app/services/albums.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 const config = require('../../config'),
   fetch = require('../../node_modules/node-fetch'),
```
Comment (igna92ts) [inline app/services/albums.js]: just import it as a module

**Review by igna92ts — CHANGES_REQUESTED** [review]

File: app/controllers/albums.js (hunk 1 above)
Comment (raparicio6) [reply app/controllers/albums.js:26]: One user can have many albums.
An album can only be from one user.
That validation is to not allow a user to buy the same album more than once.

File: app/controllers/albums.js (hunk 2 above)
Comment (raparicio6) [reply app/controllers/albums.js:28]: No. 
One user can have many albums.
An album can only be from one user.

**Pull request merged**